### ISSUE TITLE
ReferenceError: TextEncoder is not defined

### ISSUE THREAD

**Issue by dan-blank** [moonad/Formality-Core#2]
**The setup**

I went to the wiki and followed the step in "0. Installation" successfully:

\>> node -v

v8.16.0

\>> npm -v

6.4.1

\>> fmc

Formality-Core
...
Note: fmc will automatically import any local file ending in `.fmc`.

\>> lsb_release -a

No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 18.04.2 LTS
Release:	18.04
Codename:	bionic

The hello world file main.fmc:

    def main:
      "Hello, world!"

**The error**

\>> fmc main

/home/dan/opt/npm/lib/node_modules/formality-core/fm-core.js:819
  var bytes = [].slice.call(new TextEncoder("utf-8").encode(text), 0);
                                ^

ReferenceError: TextEncoder is not defined
    at text_to_term (/home/dan/opt/npm/lib/node_modules/formality-core/fm-core.js:819:33)
    at parse_term (/home/dan/opt/npm/lib/node_modules/formality-core/fm-core.js:171:14)
    at Object.parse (/home/dan/opt/npm/lib/node_modules/formality-core/fm-core.js:287:18)
    at Object.<anonymous> (/home/dan/opt/npm/lib/node_modules/formality-core/main.js:57:28)
    at Module._compile (module.js:653:30)
    at Object.Module._extensions..js (module.js:664:10)
    at Module.load (module.js:566:32)
    at tryModuleLoad (module.js:506:12)
    at Function.Module._load (module.js:498:3)
    at Function.Module.runMain (module.js:694:10)

**The error seems to appear also when called on a file that does not exist**

\>> fmc doesnotexist

/home/dan/opt/npm/lib/node_modules/formality-core/fm-core.js:819
  var bytes = [].slice.call(new TextEncoder("utf-8").encode(text), 0);
                                ^

ReferenceError: TextEncoder is not defined
    at text_to_term (/home/dan/opt/npm/lib/node_modules/formality-core/fm-core.js:819:33)
    at parse_term (/home/dan/opt/npm/lib/node_modules/formality-core/fm-core.js:171:14)
    at Object.parse (/home/dan/opt/npm/lib/node_modules/formality-core/fm-core.js:287:18)
    at Object.<anonymous> (/home/dan/opt/npm/lib/node_modules/formality-core/main.js:57:28)
    at Module._compile (module.js:653:30)
    at Object.Module._extensions..js (module.js:664:10)
    at Module.load (module.js:566:32)
    at tryModuleLoad (module.js:506:12)
    at Function.Module._load (module.js:498:3)
    at Function.Module.runMain (module.js:694:10)

**Comment by johnchandlerburnham** [comment]
Seems like your locale is not set to UTF. What does

```
$ echo $LANG
$ echo $LC_ALL
```

return for you? Try

```
$ export LC_ALL=en_US.UTF-8
```

**Comment by johnchandlerburnham** [comment]
Also you could try updating to a more recent version of `node`. If that fixes it, we should probably update the wiki's installation instructions with minimum versions

**Comment by dan-blank** [comment]
Thank you, upgrading fixed it. I am on node version v12.4.0 now.

In hindsight, I did not follow the guide exactly. Out of comfort I used the Ubuntu package manager to install node, which was why I had the version that was too old. So adding some sentence like "make sure that you are at least on version vX.X.X" would have prevented me from using the Ubuntu package manager.

I would guess that v10.16.0 (which is the lts version one gets from following the guide) is enough, but I have not verified this.
